dbt_style_guide.md

```diff
@@ -1,237 +1,415 @@
+
+- Each model should have a primary key that can identify the unique row.
+
+- The primary key of a model should be named `<object>_id`, e.g. `account_id` – this makes it easier to know what `id` is being referenced in downstream joined models.
+
+- If a surrogate key is created, it should be named `<object>_sk`.
+
+- For `base` or `staging` models, columns should be ordered in categories, where identifiers are first and date/time fields are at the end.  
+  Example:
+  ```sql
+  transformed as (
+
+      select
+
+        -- ids
+        order_id,
+        customer_id,
+
+        -- dimensions
+        order_status,
+        is_shipped,
+
+        -- measures
+        order_total,
+
+        -- metdata
+        created_at,
+        updated_at,
+        _sdc_batched_at
+
+      from source
+
+  )
+  ```
+
+- Date/time columns should be named according to these conventions:
+  - Timestamps: `<event>_at`
+    Format: UTC  
+    Example: `created_at`
+  
+  - Dates: `<event>_date`  
+    Format: Date  
+    Example: `created_date`
+
+- Booleans should be prefixed with `is_` or `has_`.
+
+- Price/revenue fields should be in decimal currency (e.g. `19.99` for $19.99; many app databases store prices as integers in cents). If non-decimal currency is used, indicate this with suffix, e.g. `price_in_cents`.
+
+- Avoid using reserved words (such as [these](https://docs.snowflake.com/en/sql-reference/reserved-keywords.html) for Snowflake) as column names.
+
+- Consistency is key! Use the same field names across models where possible.  
+Example: a key to the `customers` table should be named `customer_id` rather than `user_id`.
 
 ## CTEs
 
-For more information about why we use so many CTEs, check out [this discourse post](https://discourse.getdbt.com/t/why-the-fishtown-sql-style-guide-uses-so-many-ctes/1091).
+For more information about why we use so many CTEs, check out [this glossary entry](https://docs.getdbt.com/terms/cte).
 
-- All `{{ ref('...') }}` statements should be placed in CTEs at the top of the file
 - Where performance permits, CTEs should perform a single, logical unit of work.
-- CTE names should be as verbose as needed to convey what they do
-- CTEs with confusing or noteable logic should be commented
-- CTEs that are duplicated across models should be pulled out into their own models
-- create a `final` or similar CTE that you select from as your last line of code. This makes it easier to debug code within a model (without having to comment out code!)
-- CTEs should be formatted like this:
 
-``` sql
-with
+- CTE names should be as verbose as needed to convey what they do.
 
-events as (
+- CTEs with confusing or noteable logic should be commented with SQL comments, as you would with any complex functions.
 
-    ...
+- CTEs that are duplicated across models should be pulled out and created as their own models.
 
-),
+- CTEs and SQL within a model should follow this structure:

```

Comment:
I think this section should be moved below the next two bullet points. We don't define what an import CTE is before showing the order of it.

Reply:
+1

Reply:
+1

Reply:
[Done](https://github.com/dbt-labs/corp/pull/63/commits/a75fe2eb326e6bcc87479b1256b6b6bd84533446)